.github/workflows/test/provision-runner.yml

```diff
@@ -0,0 +1,100 @@
+      ec2-os-type:
+        description: "The os type of the ec2 instance (windows/linux)"
+        required: true
+        type: string
+      ec2-instance-type:
+        description: "The instance type for the EC2 instance"
+        required: true
+        type: string
+      aws-region:
+        description: "AWS Region"
+        required: false
+        type: string
+        default: us-east-1
+      security-group-id:
+        description: "Security group for the EC2 instance"
+        required: true
+        type: string
+      subnet-id:
+        description: "Subnet ID for the EC2 instance"
+        required: true
+        type: string
+    secrets:
+      aws-access-key-id:
+        description: "AWS Access Key ID"
+        required: true
+      aws-secret-access-key:
+        description: "AWS Secret Access Key"
+        required: true
+      github-token:
+        description: "Personal Access Token For GH"
+        required: true
+      nonadmin-password:
+        description: "Non admin user password"
+        required: true
+    outputs:
+      instance_label:
+        description: "The label of the VM that was created."
+        value: ${{ jobs.provision-ec2.outputs.label }}
+      ec2-instance-id:
+        description: "Instance id of the VM that was created."
+        value: ${{ jobs.provision-ec2.outputs.instance_id }}
+      ec2-host:
+        description: "Instance id of the VM that was created."
+        value: ${{ jobs.provision-ec2.outputs.ec2-host}}
+
+jobs:
+  provision-ec2:
+    runs-on: ubuntu-latest
+    outputs:
+      instance_id: ${{ steps.ec2.outputs.ec2-instance-id }}
+      label: ${{ steps.ec2.outputs.label }}
+      ec2-host: ${{ steps.get-ec2-host.outputs.ec2-host }}
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.aws-access-key-id }}
+          aws-secret-access-key: ${{ secrets.aws-secret-access-key }}
+          aws-region: ${{ inputs.aws-region }} # Default is us-east-1
+
+      - name: Start EC2 GitHub runner
+        id: ec2
+        uses: crunchy234/ec2-github-runner@main
+        with:
+          mode: start
+          github-token: ${{ secrets.github-token }}
+          ec2-os: ${{ inputs.ec2-os-type }}
+          ec2-instance-type: ${{ inputs.ec2-instance-type }}
+          ec2-image-id: ${{ inputs.ec2-image-id }}
+          subnet-id: ${{ inputs.subnet-id }}
+          security-group-id: ${{ inputs.security-group-id }}
+
+      - name: Get EC2 Host (Public IP/DNS)
+        id: get-ec2-host
+        run: |
+          EC2_INSTANCE_ID="${{ steps.ec2.outputs.ec2-instance-id }}"
+          EC2_INSTANCE_ID=$(echo $EC2_INSTANCE_ID | sed 's/^\[\(.*\)\]$/\1/')
+          EC2_HOST=$(aws ec2 describe-instances --instance-ids $EC2_INSTANCE_ID --query "Reservations[0].Instances[0].PublicIpAddress" --output text)
```

Comment:
we probably don't want to expose the public IP address

Reply:
Removed windows workflow for now.

f0c89c264fb75595fc777519af767d91e995e410